handler.go

```diff
@@ -0,0 +1,269 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/cyverse-de/go-mod/gotelnats"
+	"github.com/cyverse-de/p/go/svcerror"
+	"github.com/cyverse-de/p/go/user"
+	"github.com/doug-martin/goqu/v9"
+	"github.com/jmoiron/sqlx"
+	"github.com/nats-io/nats.go"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type lookupUser struct {
+	Username        string `db:"username"`
+	UserID          string `db:"id"`
+	LoginCount      uint32 `db:"login_count"`
+	Preferences     string `db:"preferences"`
+	PreferencesID   string `db:"preferences_id"`
+	Logins          []lookupLogin
+	SavedSearches   string `db:"saved_searches"`
+	SavedSearchesID string `db:"saved_searches_id"`
+}
+
+type lookupLogin struct {
+	IPAddress  sql.NullString `db:"ip_address"`
+	UserAgent  sql.NullString `db:"user_agent"`
+	LoginTime  sql.NullTime   `db:"login_time"`
+	LogoutTime sql.NullTime   `db:"logout_time"`
+}
+
+func lookupLogins(dbconn *sqlx.DB, userID string, limit, offset uint) ([]lookupLogin, error) {
+	var err error
+
+	usersT := goqu.T("users")
+	loginsT := goqu.T("logins")
+	loginsQ := goqu.From(loginsT).
+		Join(usersT, goqu.On(loginsT.Col("user_id").Eq(usersT.Col("id")))).
+		Where(usersT.Col("id").Eq(userID)).
+		Select(
+			loginsT.Col("ip_address"),
+			loginsT.Col("user_agent"),
+			loginsT.Col("login_time"),
+			loginsT.Col("logout_time"),
+		).
+		Order(loginsT.Col("login_time").Desc()).
+		Limit(limit).
+		Offset(offset)
+
+	loginsQueryString, _, err := loginsQ.ToSQL()
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := dbconn.QueryxContext(context.Background(), loginsQueryString)
```

Comment:
This should be a context passed into the function where it's called (a context with a span attached to it, specifically). Otherwise the database query will be its own isolated trace, rather than marking it as a child of whatever called into here.

(and same with other db calls, I won't leave a comment on every one of them since it'd be maddening for both of us lol)

Reply:
Thanks, should be fixed now.